spec/defines/config/server/connector_spec.rb

```diff
@@ -157,6 +141,21 @@
   end
 
   describe 'failing tests' do
+    context 'remove connector with purge_connectors' do
+      let :params do
+        {
+          connector_ensure: 'absent',
+          purge_connectors: true
+        }
+      end
+
+      it do
+        expect {
+          catalogue
+        }.to raise_error(Puppet::Error, %r{connector_ensure must be set to 'present' to use purge_connectors})
```

Comment:
Looks like the check is missing the `$` sign:
```suggestion
        }.to raise_error(Puppet::Error, %r{\$connector_ensure must be set to 'present' to use \$purge_connectors})
```